connect-inject/health_check_resource_test.go

```diff
@@ -0,0 +1,551 @@
+					Phase:  corev1.PodRunning,
+					Conditions: []corev1.PodCondition{{
+						Type:   corev1.PodReady,
+						Status: corev1.ConditionFalse,
+					}},
+				},
+			},
+			&api.AgentCheck{
+				CheckID: testHealthCheckID,
+				Status:  api.HealthCritical,
+			},
+			"",
+		},
+		{
+			"Reconcile existing object from failing to passing",
+			true,
+			api.HealthCritical,
+			&corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      testPodName,
+					Namespace: "default",
+					Labels:    map[string]string{labelInject: "true"},
+					Annotations: map[string]string{
+						annotationStatus:  "injected",
+						annotationService: testServiceNameAnnotation,
+					},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						corev1.Container{
+							Name: testPodName,
+						},
+					},
+				},
+				Status: corev1.PodStatus{
+					HostIP: "127.0.0.1",
+					Phase:  corev1.PodRunning,
+					Conditions: []corev1.PodCondition{{
+						Type:   corev1.PodReady,
+						Status: corev1.ConditionTrue,
+					}},
+				},
+			},
+			&api.AgentCheck{
+				CheckID: testHealthCheckID,
+				Status:  api.HealthPassing,
+			},
+			"",
+		},
+		{
+			"Reconcile pod not running no update",
+			false,
+			api.HealthCritical,
+			&corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      testPodName,
+					Namespace: "default",
+					Labels:    map[string]string{labelInject: "true"},
+					Annotations: map[string]string{
+						annotationStatus:  "injected",
+						annotationService: testServiceNameAnnotation,
+					},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						corev1.Container{
+							Name: testPodName,
+						},
+					},
+				},
+				Status: corev1.PodStatus{
+					HostIP: "127.0.0.1",
+					Phase:  corev1.PodFailed,
+					Conditions: []corev1.PodCondition{{
+						Type:   corev1.PodReady,
+						Status: corev1.ConditionFalse,
+					}},
+				},
+			},
+			nil,
+			"",
+		},
+	}
+	for _, tt := range cases {
+		t.Run(tt.Name, func(t *testing.T) {
+			require := require.New(t)
+
+			server, client, resource := testServerAgentResourceAndController(t, tt.Pod)
+			defer server.Stop()
+
+			// Create a passing service
+			server.AddService(t, testServiceNameReg, "passing", nil)
+			if tt.CreateHealthCheck {
+				// register the health check if ObjectCreate didnt run
+				err := registerHealthCheck(t, client, tt.InitialState)
+				require.NoError(err)
+			}
+
+			err := resource.Reconcile()
+			require.NoError(err)
+			actual := testGetConsulAgentChecks(t, client)
+			if tt.Expected == nil || actual == nil {
+				require.Equal(tt.Expected, actual)
+			} else {
+				require.Equal(tt.Expected.Status, actual.Status)
+			}
+		})
+	}
+}
+
+func TestHealthCheckHandlerStandard(t *testing.T) {
+	cases := []struct {
+		Name              string
+		CreateHealthCheck bool
+		InitialState      string
+		Pod               *corev1.Pod
+		Expected          *api.AgentCheck
+		Err               string
+	}{
```

Comment:
since this logic is now all refactored into the `reconcilePod` function, I think we're testing the same thing as above since Upsert just calls `reconcilePod`. Maybe we can have tests specifically for the `reconcilePod`, and then test `Reconcile` and `Upsert` just for the behavior they have that is unique to them?